pymeasure/adapters/serial.py

```diff
@@ -103,9 +103,23 @@ def _read_bytes(self, count, break_on_termchar, **kwargs):
             return self.connection.read_until(self.read_termination.encode(),
                                               count if count > 0 else None,
                                               **kwargs)
+        elif count >= 0:
+            return self.connection.read(count, **kwargs)
         else:
-            # At -1 we read a very large number of bytes, which can be considered the whole buffer.
-            return self.connection.read(1e99 if count == -1 else count, **kwargs)
+            # For -1 we empty the buffer completely
+            return self._read_bytes_until_timeout()
+
+    def _read_bytes_until_timeout(self, **kwargs):
+        """Read from the serial until a timeout occurs, regardless of how many bytes."""
+        # `Serial.readlines()` sounds appropriate instead but it turns out it has an
```

Comment:
That comment is in the PR discussion (we could add it to the issue as well), but does not need to be in the source code.

Reply:
It's useful context, though ("comment why, not what") -- reading without knowing about the issue, the first thing one could be wondering why pymeasure is reimplementing `Serial.readlines`.

Reply:
I agree with @bilderbuchi , the 'why' is leading here. Reduced comment to one line and linked this PR.